### PR TITLE
#27 Added initial async interfaces

### PR DESCRIPTION
**Pull request by JigarJoshi**
#27 

 - Added async interfaces.
 - Removed AlterCollection API (as it is going to get removed from server soon).
 
 TODO:
 - refactor Fields for read and update.

### CODE REVIEW

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+   * inserts a single document to the collection
+   *
+   * @param document
+   * @return a future to the {@link InsertResponse}
+   * @throws TigrisDBException
+   */
+  CompletableFuture<InsertResponse> insert(T document) throws TigrisDBException;
+
+  /**
+   * Inserts the documents if they don't exist already (matched by primary key), replaces them
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java:96]: remove `(matched by primary key)` as it applies to all write apis so not adding any benefit adding here.

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+   * otherwise.
+   *
+   * @param documents list of documents to replace
+   * @param insertOrReplaceRequestOptions replace option
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java:100]: remove replace.

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+  /**
+   * @param filter
+   * @param fields
+   * @param updateRequestOptions
+   * @return a future to the {@link UpdateResponse}
+   * @throws TigrisDBException
+   */
+  CompletableFuture<UpdateResponse> update(
+      TigrisFilter filter, List<Field<?>> fields, UpdateRequestOptions updateRequestOptions)
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java:127]: is this Field class same as read? They should be different.

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncDatabase.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  /**
+   * Alters a collection under current database.
+   *
+   * @param collectionName name of the collection
+   * @param schema schema of the collection
+   * @param collectionOptions collection option
+   * @return the future to the {@link AlterCollectionResponse}
+   * @throws TigrisDBException in case of an error.
+   */
+  CompletableFuture<AlterCollectionResponse> alterCollection(
```
Comment (himank) [inline client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncDatabase.java:60]: remove it and just have createCollection for now as we will be remove alter from the server side as well.

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java (hunk 1 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java]: removed from all docs.

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncDatabase.java (hunk 4 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncDatabase.java]: removed AlterCollection API from client.

File: client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java (hunk 3 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/TigrisAsyncCollection.java:125]: I will refactor this in next PR.

**Review by himank — APPROVED** [review]

**Pull request merged**